### PR TITLE
[keba] Split channel ENABLED to USER and SYSTEM channels

### PR DESCRIPTION
**Pull request by maihacke**
This PR split the two channels for enabled_system and enabled_user state of the wallbox.
See https://github.com/openhab/openhab-addons/issues/15521

### CODE REVIEW

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml [hunk 1]
```diff
@@ -119,7 +120,7 @@
 		<item-type>Switch</item-type>
 		<label>Enabled</label>
 		<description>Activation state of the wallbox</description>
-		<state readOnly="false"></state>
+		<state readOnly="true"></state>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml:123]: It is not read only, at least for enableduser at commands are handled on that channel.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml [hunk 2]
```diff
@@ -9,7 +9,8 @@
 		<description>A KeContact EV Charging Station</description>
 
 		<channels>
-			<channel id="enabled" typeId="enabled"/>
+			<channel id="enabledsystem" typeId="enabled"/>
+			<channel id="enableduser" typeId="enabled"/>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml:13]: Please update the documentation to describe these 2 channels.
Please provide upgrade instructions (XML file) for an automatic update of the existing things for users suing managed things.
Add a description here for each channel to override the generic description of the channel type (enabled) and then regenerate default translations property file.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml (hunk 1 above)
Comment (maihacke) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml:123]: Fixed

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml:13]: > --> Done

Maybe you did it but you did not push your changes if you did it.

Comment (maihacke) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/thing/kecontact.xml:13]: Strange, maybe I just forgot or it was interrupted and I did notice... sorry. :astonished:

File: bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java [hunk 3]
```diff
@@ -325,11 +325,25 @@ protected void onData(ByteBuffer byteBuffer) {
...
+                        switch (state) {
+                            case 1: {
+                                updateState(CHANNEL_ENABLED_USER, OnOffType.ON);
+                                break;
+                            }
+                            default: {
+                                updateState(CHANNEL_ENABLED_USER, OnOffType.OFF);
                                 break;
                             }
                         }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java:349]: For your information, you could replace all this block by just:
```
updateState(CHANNEL_ENABLED_USER, OnOffType.from(entry.getValue().getAsInt() == 1));
```

Same for block just before.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+<update:update-descriptions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:update="https://openhab.org/schemas/update-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/update-description/v1.0.0 https://openhab.org/schemas/update-description-1.0.0.xsd">
+
+	<thing-type uid="keba:kecontact">
+		<instruction-set targetVersion="1">
+			<remove-channel id="enabled"/>
+			<add-channel id="enabledsystem">
+				<type>keba:enaledsystem</type>
+				<label>Enabled (System)</label>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:11]: No need to set the label.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+			</add-channel>
+			<add-channel id="enableduser">
+				<type>keba:enaleduser</type>
+				<label>Enabled (User)</label>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:15]: No need to set the label

**Review by lolodomo — CHANGES_REQUESTED** [review]
Just one remark regarding upgrade instructions + a proposal for a simplification.
Do the simplification only if you like it.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml (hunk 4 above)
Comment (maihacke) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:11]: Labels is set for all channels in this binding. I assume it's good to have no exception here for uniformity.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml (hunk 5 above)
Comment (maihacke) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:15]: Labels are set for all channels in this binding. I assume it's good to have no exception here for uniformity.

File: bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java (hunk 3 above)
Comment (maihacke) [reply bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java]: Good catch, that was copy & paste from the original code. I like the simplification and updated the PR.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml (hunk 4 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:11]: Label of course is set in thing description. Here we are in upgrade instructions and redefining a label is useless.

Comment (jlaur) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:11]: It will also overwrite any localized label.

Comment (maihacke) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml]: Label removed.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml (hunk 5 above)
Comment (maihacke) [reply bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml]: Label removed

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+<update:update-descriptions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:update="https://openhab.org/schemas/update-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/update-description/v1.0.0 https://openhab.org/schemas/update-description-1.0.0.xsd">
+
+	<thing-type uid="keba:kecontact">
+		<instruction-set targetVersion="1">
+			<remove-channel id="enabled"/>
+			<add-channel id="enabledsystem">
+				<type>keba:enaledsystem</type>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:10]: A "b" is missing.

File: bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+			</add-channel>
+			<add-channel id="enableduser">
+				<type>keba:enaleduser</type>
```
Comment (lolodomo) [inline bundles/org.openhab.binding.keba/src/main/resources/OH-INF/update/kecontact-enabled.xml:13]: A "b" is missing.

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**